docs/proposals/csi-deployment-crd.md

```diff
@@ -0,0 +1,269 @@
+# Proposal:  CSI deployment CRD
+
+Last updated: 05.03.20
+
+## Abstract
+
+To deploy CSI component with operator, it's necessary to provide CRD so operator can deploy CSI programmable using its configuration 
+
+## Background
+
+Currently CSI is deployed using helm charts and value
+
+## Proposal
+
+Example of CRD:
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: csi-baremetal.dell.com
+  labels:
+    app.kubernetes.io/name: csi-baremetal-operator
+spec:
+  group: csi-baremetal.dell.com
+  names:
+    kind: CSI
+    listKind: CSIList
+    plural: csi
+    singular: csi
+  scope: Cluster
+  versions:
+    - name: v1beta1
+      served: true
+      storage: true
+```
+
+Example of CR:
+```
+apiVersion:  csi-baremetal.dell.com/v1beta1
+kind: CSI
+metadata:
+  name: csi
+  namespace: default
+  labels:
+    app.kubernetes.io/name: csi-baremetal
+    app.kubernetes.io/managed-by: csi-baremetal-operator
+    app.kubernetes.io/version: 1.0.0
+spec:
+  driver:
+    controller:
+      image:
+        registry: asdrepo.isus.emc.com:9042
+        name: csi-baremetal-driver
+        pullPolicy: Always
+        tag: green
+      log:
+        format: text
+        level: info
+      sidecars:
+        - name: csi-provisioner
+          image:
+            name: csi-provisioner
+            registry: asdrepo.isus.emc.com:9042
+            pullPolicy: Always
+            tag: v1.6.0
+        - name: csi-attacher
+          image:
+            name: csi-attacher
+            registry: asdrepo.isus.emc.com:9042
+            pullPolicy: Always
+            tag: v1.0.1
+        - name: csi-resizer
+          image:
+            name: csi-resizer
+            registry: asdrepo.isus.emc.com:9042
+            pullPolicy: Always
+            tag: v1.1.0
+      testEnv: false
+    node:
+      driveMgr:
+        image:
+          name: csi-baremetal-basemgr
+          registry: asdrepo.isus.emc.com:9042
+          pullPolicy: Always
+          tag: green
+        endpoint: tcp://localhost:8888
+        deployConfig: false
+        amountOfLoopback: 3
+        sizeOfLoopback: 101Mi
+      image:
+        registry: asdrepo.isus.emc.com:9042
+        pullPolicy: Always
+        tag: green
+      log:
+        format: text
+        level: info
+      sidecars:
+        - name: csi-node-driver-registrar
+          image:
+            name: csi-node-driver-registrar
+            registry: asdrepo.isus.emc.com:9042
+            pullPolicy: Always
+            tag: v1.0.1-gke.0
+      testEnv: false
+    metrics:
+      path: /metrics
+      port: 8787
+    logReceiver:
+      name: fluent-bit
+      image:
+        name: fluent-bit
+        registry: asdrepo.isus.emc.com:9042
+        pullPolicy: Always
+        tag: shippable
+    deployAlertsConfig: false
+    nodeSelector:
```

Comment:
`nodeSelector` should be a common setting. Is `os=linux` just an example? We can't have default value here...

Reply:
It's just an example